iml-gui/crate/src/components/stratagem/inode_table.rs

```diff
@@ -0,0 +1,199 @@
+
+#[derive(serde::Deserialize, serde::Serialize, Debug, Eq, PartialEq, Ord, PartialOrd, Clone)]
+pub(crate) struct INodeCount {
+    count: u32,
+    size: i64,
+    uid: String,
+    timestamp: i64,
+}
+
+pub struct Model {
+    fs_name: String,
+    inodes: Vec<INodeCount>,
+    cancel: Option<oneshot::Sender<()>>,
+    request_controller: Option<RequestController>,
+    last_known_scan: Option<String>,
+}
+
+impl Model {
+    pub fn new(fs_name: &String) -> Self {
+        Model {
+            fs_name: fs_name.clone(),
+            inodes: Default::default(),
+            cancel: Default::default(),
+            request_controller: Default::default(),
+            last_known_scan: Default::default(),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub enum Msg {
+    FetchInodes,
+    InodesFetched(fetch::ResponseDataResult<InfluxResults>),
+    Noop,
+}
+
+#[derive(serde::Deserialize, Clone, Debug)]
+pub struct InfluxSeries {
+    #[serde(skip)]
+    name: String,
+    #[serde(skip)]
+    columns: Vec<String>,
+    values: Vec<(i64, String, u32, i64)>,
+}
+
+#[derive(serde::Deserialize, Clone, Debug)]
+pub struct InfluxResult {
+    #[serde(skip)]
+    statement_id: u16,
+    series: Option<Vec<InfluxSeries>>,
+}
+
+#[derive(serde::Deserialize, Clone, Debug)]
+pub struct InfluxResults {
+    results: Vec<InfluxResult>,
+}
+
+#[derive(Debug, Clone)]
+enum InodeError {
+    Cancelled(futures::channel::oneshot::Canceled),
+    FailedFetch(seed::fetch::FailReason<InfluxResults>),
+}
+
+impl std::fmt::Display for InodeError {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match *self {
+            InodeError::Cancelled(ref err) => write!(f, "{}", err),
+            InodeError::FailedFetch(ref err) => write!(f, "{:?}", err),
+        }
+    }
+}
+
+impl std::error::Error for InodeError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match *self {
+            InodeError::Cancelled(ref err) => Some(err),
+            InodeError::FailedFetch(_) => None,
+        }
+    }
+}
+
+impl From<futures::channel::oneshot::Canceled> for InodeError {
+    fn from(err: futures::channel::oneshot::Canceled) -> Self {
+        InodeError::Cancelled(err)
+    }
+}
+
+impl From<seed::fetch::FailReason<InfluxResults>> for InodeError {
+    fn from(err: seed::fetch::FailReason<InfluxResults>) -> Self {
+        InodeError::FailedFetch(err)
+    }
+}
+
+pub(crate) fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg, GMsg>) {
+    match msg {
+        Msg::FetchInodes => {
+            model.cancel = None;
+            let url:String = format!("influx?db=iml_stratagem_scans&epoch=ns&q=SELECT%20counter_name,%20count,%20size%20FROM%20stratagem_scan%20WHERE%20group_name=%27user_distribution%27%20and%20fs_name=%27{}%27%20limit%2020", &model.fs_name);
+            let request =
+                seed::fetch::Request::new(url).controller(|controller| model.request_controller = Some(controller));
```

Comment:
We shouldn't need to deal with request_controller I think.

Reply:
Removed.